modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/clients/APIPublisherRestClient.java

```diff
@@ -584,14 +556,131 @@ public HttpResponse changeAPILifeCycleStatusToPublish(APIIdentifier apiIdentifie
+
+		try {
+			checkAuthentication();
+			return HttpRequestUtil.doGet(backEndUrl + "/publisher/site/pages/tiers.jag", requestHeaders);
+		} catch (Exception e) {
+			throw new APIManagerIntegrationTestException("Exception when retrieving the Tier Permissions page", e);
+		}
+
+	}
+
+	/**
+	 * Retrieve the API Manage Page
+	 *
+	 * @param apiName  - Name of the API.
+	 * @param provider - Name of the API Provider.
+	 * @param version  - Version of the API.
+	 * @return HttpResponse - Response that contains the API Manage Page
+	 * @throws APIManagerIntegrationTestException - Exception throws from checkAuthentication() method and
+	 *                                            HttpRequestUtil.doGet() method call
+	 */
+	public HttpResponse getAPIManagePage(String apiName, String provider, String version) throws APIManagerIntegrationTestException {
+		try {
+			checkAuthentication();
+			return HttpRequestUtil.doGet(backEndUrl + "/publisher/manage?name=" + apiName + "&version=" + version +
+					"&provider=" + provider, requestHeaders);
+		} catch (Exception e) {
+			throw new APIManagerIntegrationTestException("Exception when retrieving the API Manage page", e);
+		}
+
+	}
+
+
+	/**
+	 * Retrieve the API Information Page
+	 *
+	 * @param apiName  - Name of the API.
+	 * @param provider - Name of the API Provider.
+	 * @param version  - Version of the API.
+	 * @return HttpResponse - Response that contains the API Information Page
+	 * @throws APIManagerIntegrationTestException - Exception throws from checkAuthentication() method and
+	 *                                            HttpRequestUtil.doGet() method call
+	 */
+	public HttpResponse getAPIInformationPage(String apiName, String provider, String version)
+			throws APIManagerIntegrationTestException {
+		try {
+			checkAuthentication();
+			return HttpRequestUtil.doGet(backEndUrl + "/publisher/info?name=" + apiName + "&version=" + version +
+					"&provider=" + provider, requestHeaders);
+		} catch (Exception e) {
+			throw new APIManagerIntegrationTestException("Exception when retrieving the API Information page", e);
+		}
+
+	}
+
+	/**
+	 * Adding a Document to a API
+	 *
+	 * @param apiName     - Name of the API.
+	 * @param version     - Version of the API.
+	 * @param provider    - Name of the API Provider.
+	 * @param docName     - Name of the Document
+	 * @param docType     - Document Type
+	 * @param sourceType  - Source Type
+	 * @param docUrl      - Document URL
+	 * @param summary     - Document summary
+	 * @param docLocation - Document Location
+	 * @return HttpResponse - Response  with Document adding result.
+	 * @throws APIManagerIntegrationTestException - Exception throws from checkAuthentication() method and
+	 *                                            HttpRequestUtil.doPost() method call
+	 */
+	public HttpResponse addDocument(String apiName, String version, String provider, String docName, String docType,
```

Comment:
Method is taking too much parameters.. Shouldn't we introduce a bean to hold all values and pass it.